lib/cli/src/Cardano/CLI.hs

```diff
@@ -1128,14 +1133,25 @@ decodeError bytes = do
     Aeson.parseMaybe (Aeson.withObject "Error" (.: "message")) obj
 
 -- | Resolve '~' or '$HOME' in a 'FilePath' to their actual system value
-resolveHomeDir :: FilePath -> IO FilePath
+resolveHomeDir
+    :: forall t. Typeable t
+    => FilePath
+    -> IO FilePath
 resolveHomeDir dir = do
+    let namespace = typeRepFilePath $ typeRep $ Proxy @t
     homeDir <- T.pack <$> getHomeDirectory
     return
         $ T.unpack
         $ T.replace "$HOME" homeDir
         $ T.replace "~" homeDir
```

Comment:
Had I have seen this before, I would have objected.
1. This substitution should be the job of the command shell.
2. It looks weird on windows.
3. The default of `~/.cardano-wallet` is inconsiderate to users because it clutters their home directory. Better to use `$XDG_DATA_HOME`

The [xdg-basedir](http://hackage.haskell.org/package/xdg-basedir) package will help.